src/utils/auth.js

```diff
@@ -0,0 +1,40 @@
+const BASE_URL = "https://api.news-explorer-ofir.students.nomoredomainssbs.ru";
+
+const handleResponse = async (res) => {
+  if (res.ok) {
+    return res.json();
+  } else {
+    return Promise.reject(await res.json());
+  }
+};
+
+const register = async ({email, password, name}) => {
+  const res = await fetch(`${BASE_URL}/signup`, {
+    method: "POST",
+    headers: {"Content-Type": "application/json"},
+    body: JSON.stringify({email, password, name})
+  });
+  return handleResponse(res);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(handleResponse)
  }

```

And now you can safely replace `fetch` with `request`. And you don’t need to change anything else.